javalib/src/main/scala/java/io/BufferedReader.scala

```diff
@@ -127,8 +128,25 @@ class BufferedReader(in: Reader, sz: Int) extends Reader {
   def lines(): Stream[String] =
     lines(false)
 
-  private[java] def lines(closeAtEnd: Boolean): Stream[String] =
-    new WrappedScalaStream(toScalaStream(closeAtEnd), None)
+  private[java] def lines(closeAtEnd: Boolean): Stream[String] = {
+    val spliter =
+      new Spliterators.AbstractSpliterator[String](Long.MaxValue, 0) {
+        def tryAdvance(action: Consumer[_ >: String]): Boolean = {
+          Option(readLine()) match {
```

Comment:
In this single case `Option` only protects us from null, however we might match on `null` | `line` here directly to skip unnecessary boxing

Reply:
@Wojciech 
Thank you for taking a rapid overview look at this PR so soon after it was created.

This is a good suggestion.  Every cycle saved, especially in library code, helps.

I will make the correction, once I get thru CI.

Sorry the eyesores you are bound to encounter on detailed review. 


I have made several refinement passes and hope to continue that once
I have confidence in correctness across hardware and Scala and Java versions.

I always like improving both performance in code and my Scala writing skills,
so call out where iterative improvement is indicated.  Thank you, and I hope
that I do not take up too much of your time.